src/http.cpp

```diff
@@ -998,6 +1002,10 @@ Request::handle_response_body(const asio::error_code& ec, const size_t bytes)
         conn_->async_read(content_length - response_.body.size(),
             std::bind(&Request::handle_response_body, this, std::placeholders::_1, std::placeholders::_2));
     }
+    else if (response_.status_code >= 400 and response_.status_code <= 500)
+    {
+        terminate(asio::error::operation_aborted);
```

Comment:
response with body should be properly handled regardless of the HTTP response code

Reply:
@aberaud I think the body cases are well handled in the lines before as tested with [http_errors.cpp](https://github.com/binarytrails/various/blob/master/cpp/opendht/http_errors.cpp) because this condition is in else if after all possible body reads